dowhy/causal_model.py

```diff
@@ -191,6 +192,20 @@ def identify_effect(
         if estimand_type is None:
             estimand_type = self._estimand_type
 
+        all_graph_variables = set.union(
+            set(self.get_common_causes()),
+            set(self.get_effect_modifiers()),
+            set(self.get_instruments()),
+        )
+        unobserved_graph_variables = all_graph_variables.difference(self._data.columns)
+
+        if unobserved_graph_variables:
+            warnings.warn(
+                f"The following graph variables were missing from the observed data: "
```

Comment:
From the wording, it's not clear whether this is an ok or not ok thing.  To resolve this, I'd suggest giving a little more guidance.  For larger graphs, we might also want to add counts too.

Maybe something like:  "The graph defines N variables.  K were found in the dataset and will be analyzed as observed variables.  N-K were not found in the dataset and will be analyzed as unobserved variables.  The observed variables are: ...  The unobserved variables are ...  If this matches your expectations for observations, please continue. If you expected any of these unobserved variables to be in the dataframe, please check for typos."

Maybe also "there are an additional Z variables in the dataset that are not in the graph".

Reply:
Regarding your comment about whether the warnings should be visible or not by default ... how about we have a short message that says "N variables are assumed unobserved because they are not in the dataset.  Pass/set the ____ flag for additional details".

Reply:
I've tried to adapt the logging to resolve your comments. There are now two levels to the output:
* A high level warning, emitted using `warnings.warn` that points out the issue and suggests additional details are available when configuring `logging` levels.
* A lower level, more detailed warning, emitted using `self.logger.warn` that follows the format you describe above.

Regarding your last point *there are an additional Z variables in the dataset that are not in the graph*, I decided to treat this under a separate conditional (as it may occur in cases without unobserved graph variables). This seems to me like it may be less likely to unveil a user issue and will only be visible when `logging` is configured to at least the `WARNING` level.

Let me know your thoughts.